packages/x-data-grid/src/components/virtualization/GridVirtualScrollerRenderZone.tsx

```diff
@@ -48,13 +49,13 @@ const GridVirtualScrollerRenderZone = React.forwardRef<
 
   return (
     <VirtualScrollerRenderZoneRoot
-      ref={ref}
       className={clsx(classes.root, className)}
       ownerState={rootProps}
       style={{
         transform: `translate3d(0, ${offsetTop}px, 0)`,
       }}
       {...other}
+      ref={ref}
```

Comment:
This change is wrong no? The ref needs to arrive before `{...other}`: otherwise, it goes into the slow branch

<img width="1148" alt="SCR-20241229-cgoo" src="https://github.com/user-attachments/assets/fd09cc3b-e259-455b-8924-22cec43099ea" />

[Source](https://babeljs.io/repl#?browsers=&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=DwNQlgTgLgrghgGwMoGMIHsEIKYQErYB2AJrgFrqHZ7rpQBQABIygnAM7sBycAttgF4A3q3YAPABSsO7bOwB0GOgBoWbTj34BKAL5NG6AO5UISKHCiChSqAAUMAB3Z7m7KAE8cwofuZQIcITsAGboELwAXIwABv6B7GyWAMzEEgAMqgAkQujBwbJQACroDjoOYqppWtHK-joujELyzXQAFrgNENjBwl3BegD0AHxAA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Ctypescript&prettier=true&targets=Node-14&version=7.26.4&externalPlugins=%40babel%2Fplugin-transform-class-properties%407.24.6%2C%40babel%2Fplugin-transform-class-static-block%407.24.6%2C%40babel%2Fplugin-transform-runtime%407.25.7&assumptions=%7B%7D)

See https://github.com/facebook/react/blob/50f00fd876b0b92b243cd8b54a222f9577446392/packages/react/src/jsx/ReactJSXElement.js#L302C7-L302C15

Reply:
Na, nevermind. I'm confusing ref and key.